### DISCUSSION TITLE
Telemetry: message intents on first launch are unclear and feel ominous

### DISCUSSION THREAD

**Discussion by Oliquae** [withastro/roadmap#746] [Proposal]
# Summary

Supposing that people might not actively follow Astro's news and about Telemetry's presence in Astro, the message shown on the first launch does not feel reassuring about how data are used.

# Background & Motivation

I started a new Astro project, unaware that Telemetry is used in Astro. I already have a project using it, so it was the first time I saw this message:

```bash
╭─[$HOST] as $USER in /**/new-project-name 
╰─> npm run dev

> new-project-name@0.0.1 dev
> astro dev

  ◆ Astro collects completely anonymous usage data.
     This optional program helps shape our roadmap.
     Run `npm run astro telemetry disable` to opt-out.
     Details: https://astro.build/telemetry
```

Given that I didn't have the whole context, my first feeling wasn't positive and all, because it really gave "*Big Money Company is looking at your computer and how you use it*", just like the big ones. Reading the doc to have more extensive information about how data would be used would give nothing but defiance about Astro.

It did make me aware the the presence of telemetry in present web tools. The principle itself may be disputable, even though the intentions are good and just functional for Astro's roadmap. **But the opt-out policy by default does not help.**

I heard that NuxtJS has a more opt-in approach that seems more middle ground.

**May the telemetry be used in Astro, a bit more transparency about intentions would be welcome and reassuring.**

This has been discussed in Discord Community.

# Goals

Believing Telemetry intentions are more about functionality, prioritization, and understanding how the software is used, it would be good to either:

- During project creation, have Houston explain in the terminal that telemetry could be used, maybe with an example, and ask for its use in the project with a "yes" or "no" answer;
- During the first launch, have a more explicit and probably more inviting message about how telemetry is used;
    - Probably update the documentation page with an example of the content in a formatted block;
- Both of the previous points;
- Display a public dashboard with anonymized data and statistics so we know how it is used.

**Comment by cdtut** [comment]
It decrease trust in Astro and maintainers behind it.

**Comment by NikolaRHristov** [comment]


**Reply by delucis** [reply]
The GDPR only requires opt-in data collection for personally identifiable information (PII). I’m not sure for other frameworks, but Astro very intentionally collects limited information for this reason as documented on the [Astro telemetry page](https://astro.build/telemetry/).

**Reply by NikolaRHristov** [reply]
My bad.

**Comment by ayo-run** [comment]
Hi all! Also recently just found out about the telemetry as I have not been able to follow Astro for a while. Tested a few runs and, there is no message shown at any point in creating a new project via `npm create astro@latest` and/or running `astro dev` (unlike in @Palingenae original post/discussion). Was the message on `astro dev` removed? That seems a step back from what was suggested in this discussion.

**Reply by ayo-run** [reply]
🙏 If I may add to the suggestions, it would be great to have a per-project opt-out (via config?) so every contributor and CI environment to the project is opted-out. If a project owner is concerned about telemetry for whatever reason, this gives us an easy control instead of the current way where we have to inform contributors, run `astro telemetry disable` per npm user (per machine?) and set the `ASTRO_TELEMETRY_DISABLED=1` env variable per environment.

**Comment by randomguy-2650** [comment]
> The GDPR only requires opt-in data collection for personally identifiable information (PII). I’m not sure for other frameworks, but Astro very intentionally collects limited information for this reason as documented on the [Astro telemetry page](https://astro.build/telemetry/).

This is _technically_ inaccurate under EU law. The GDPR does not use the term “PII” (which I believe is a U.S. legal concept); it uses **Personal Data**. Under Art. 4(1) and Recital 26, information is personal data if it relates to an identifiable person.

- **Pseudonymization ≠ Anonymization:** According to the CJEU Breyer Case (C-582/14), dynamic IP addresses are personal data. Since telemetry is sent from a user’s machine, the IP is processed at ingestion. Even if the data is “sanitised” later, the collection process itself involves personal data.
- **Device fingerprinting:** The list of “anonymous” data (CPU count, OS version, integrations list, CI environment) constitutes a High‐Entropy Fingerprint. The ePrivacy Directive (the “Cookie Law”) and Art. 5(3) require prior consent for accessing or storing information on a user’s device for non‐essential purposes (like telemetry), regardless of whether it’s “PII” or not.

By making telemetry opt‐out, the framework relies on GDPR Art. 6(1)(f) (Legitimate Interest). However, using this for product roadmaps and such often fails the Necessity and Balancing tests because it is not strictly necessary for the software to function, and it ignores the rights of privacy‐conscious users who have already opted out.

<sup>I’m not a lawyer, but this information should be accurate as I had to deal with GDPR for an open‐source project extensively (i.e., fully).</sup>
